File: Python/AbstractFactory.py
import abc;

### DEFINING INTERFACES ###

class AbstractFactory():
	@abc.abstractMethod
	def createProductA(self):
		pass;
	
	@abc.abstractMethod
	def createProductB(self):
		pass;

class ProductA():
	@abc.abstractMethod
	def greetings(self):
		pass;

class ProductB():
	@abc.abstractMethod
	def audience(self):
		pass;


#--------MAIN--------------

class Client():
	def __new__(self, a, b):
		self.a = a;
		self.b = b;
	
	#?????
	def main(self, String[] args) {
		a.greetings();
		print(", ");
		b.audience();
		print("!");

### PRODUCTS ###

class Hello(ProductA):
	def greetings(self):
		print("Hello");

class World(ProductB):
	def audience(self):
		print("World");

class ToMe(ProductB):
	def audience(self):
		print("To me");

### Factories ###

class HappyFactory(AbstractFactory):
	def createProductA(self):
		return Hello();
	
	def createProductB(self):
		return World();


class GrumpyFactory(AbstractFactory):
	def createProductA():
		return Hello();
	
	def createProductB():
		return ToMe();

File: Java/AbstractFactory.java
interface AbstractFactory {
	public ProductA createProductA();
	public ProductB createProductB();
}

interface ProductA {
	public void greetings();
}

interface ProductB {
	public void audience();
}

//----------------------------

public class Client {
	
	private ProductA a;
	private ProductB b;
	
	public Client(ProductA a, ProductB b) {
		this.a = a;
		this.b = b;
	}
	
	public static void main(String[] args) {
		a.greetings();
		System.out.print(", ");
		b.audience();
		System.out.print("!");
	}
}

public class Hello implements ProductA {
	public void greetings() {
		System.out.print("Hello")
	}
}

public class World implements ProductB {
	public void audience() {
		System.out.print("World")
	}
}

public class ToMe implements ProductB {
	public void audience() {
		System.out.print("To me")
	}
}

public class HappyFactory implements AbstractFactory {
	public ProductA createProductA() {
		return new Hello();
	}
	
	public ProductB createProductB() {
		return new World();
	}
}

public class GrumpyFactory implements AbstractFactory {
	public ProductA createProductA() {
		return new Hello();
	}
	
	public ProductB createProductB() {
		return new ToMe();
	}
}